src/is-supported.ts

```diff
@@ -2,8 +2,13 @@ import { mimeTypeForCodec } from './utils/codecs';
 import { getMediaSource } from './utils/mediasource-helper';
 import type { ExtendedSourceBuffer } from './types/buffer';
 
-function getSourceBuffer(): typeof self.SourceBuffer {
-  return self.SourceBuffer || (self as any).WebKitSourceBuffer;
+function getSourceBuffer(
+  preferManagedSourceBuffer = true,
+): typeof self.SourceBuffer {
+  const msb =
+    (preferManagedSourceBuffer || !self.SourceBuffer) &&
+    ((self as any).ManagedSourceBuffer as undefined | typeof self.SourceBuffer);
+  return msb || self.SourceBuffer || (self as any).WebKitSourceBuffer;
```

Comment:
I'd be willing to merge with the following change. [See previous comment for reasoning](https://github.com/video-dev/hls.js/pull/6846/files#r1847086787):
```suggestion
function getSourceBuffer(): typeof self.SourceBuffer {
  return self.SourceBuffer || (self as any).ManagedSourceBuffer || (self as any).WebKitSourceBuffer;
```

Reply:
@niklaskorz would you be willing to make the requested changes?

Reply:
This won't make it into the next release without changes. Let me know if you'd like to update or close the issue.

Reply:
Sorry for the very long radio silence @robwalch, I'll take another look at this today and get back to you